stats_collector/prometheus.go

```diff
@@ -0,0 +1,388 @@
+	decodeDiskEncounter.WithLabelValues(status, message).Inc()
+}
+
+func (col *promCollector) IncDecodeQuest(status, message string) {
+	decodeQuest.WithLabelValues(status, message).Inc()
+}
+
+func (col *promCollector) IncDecodeSocialActionWithRequest(status, message string) {
+	decodeSocialActionWithRequest.WithLabelValues(status, message).Inc()
+}
+
+func (col *promCollector) IncDecodeGetFriendDetails(status, message string) {
+	decodeGetFriendDetails.WithLabelValues(status, message).Inc()
+}
+
+func (col *promCollector) IncDecodeSearchPlayer(status, message string) {
+	decodeSearchPlayer.WithLabelValues(status, message).Inc()
+}
+
+func (col *promCollector) IncDecodeGMO(status, message string) {
+	decodeGMO.WithLabelValues(status, message).Inc()
+}
+
+func (col *promCollector) AddDecodeGMOType(typ string, value float64) {
+	decodeGMOType.WithLabelValues(typ).Add(value)
+}
+
+func (col *promCollector) IncDecodeStartIncident(status, message string) {
+	decodeStartIncident.WithLabelValues(status, message).Inc()
+}
+
+func (col *promCollector) IncDecodeOpenInvasion(status, message string) {
+	decodeOpenInvasion.WithLabelValues(status, message).Inc()
+}
+
+func (col *promCollector) AddPokemonStatsResetCount(area string, val float64) {
+	pokemonStatsResetCount.WithLabelValues(area).Add(val)
+}
+
+func (col *promCollector) IncPokemonCountNew(area string) {
+	pokemonCountNew.WithLabelValues(area).Inc()
+}
+
+func (col *promCollector) IncPokemonCountIv(area string) {
+	pokemonCountIv.WithLabelValues(area).Inc()
+}
+
+/*
+func (col *promCollector) IncPokemonCountShiny(area string) {
+	pokemonCountShiny.WithLabelValues(area).Inc()
+}
+
+func (col *promCollector) IncPokemonCountShundo(area string) {
+	pokemonCountShundo.WithLabelValues(area).Inc()
+}
+
+func (col *promCollector) IncPokemonCountSnundo(area string) {
+	pokemonCountSnundo.WithLabelValues(area).Inc()
+}
+*/
+
+func (col *promCollector) IncPokemonCountHundo(area string) {
+	pokemonCountHundo.WithLabelValues(area).Inc()
+}
+
+func (col *promCollector) IncPokemonCountNundo(area string) {
+	pokemonCountNundo.WithLabelValues(area).Inc()
+}
+
+func (col *promCollector) UpdateVerifiedTtl(area geo.AreaName, seenType null.String, expireTimestamp null.Int) {
+	remainingTtlMin := (expireTimestamp.ValueOrZero() - time.Now().Unix()) / 60
+	var seenTypeStr = seenType.String
+	above30 := "0"
+
+	if remainingTtlMin < 0 {
+		return
+	}
+
+	// set above30 when TTL is over 30 minutes
+	// depending on the route times can be unreliable
+	if remainingTtlMin > 30 {
+		above30 = "1"
+	}
+
+	verifiedPokemonTTL.WithLabelValues(area.String(), seenTypeStr, above30).Add(float64(remainingTtlMin))
+	verifiedPokemonTTLCounter.WithLabelValues(area.String(), seenTypeStr, above30).Inc()
+}
+
+func (col *promCollector) UpdateRaidCount(areas []geo.AreaName, raidLevel int64) {
+	processed := make(map[string]bool)
+	for _, area := range areas {
+		if !processed[area.String()] {
```

Comment:
I think this is cut & paste code, but the duplicate call to .String() here may not be able to be optimised by the compiler (but we know it is the same value)

Reply:
Yeah, those functions were copied as-is and I didn't pay much attention to them. I agree with only calling String() once. This has been updated now in all of the places...

Reply:
https://github.com/UnownHash/Golbat/compare/be15459f094400fe5ea9bdc5b1ef238238a71542..a23d1a5b9b56e7caada81816748d55e8f5da1d2b